src/common/storage/src/config.rs

```diff
@@ -474,3 +478,60 @@ impl ShareTableConfig {
         }
     }
 }
+
+#[derive(Clone, Default, PartialEq, Eq, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct CatalogConfig {
+    #[serde(flatten)]
+    pub catalogs: HashMap<String, CatalogDescription>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "type")]
+pub enum CatalogDescription {
+    #[serde(rename = "hive")]
+    Hive(HiveCatalogConfig),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "protocol")]
+pub enum ThriftProtocol {
+    #[serde(rename = "binary")]
+    Binary,
+    // Compact,
+}
+
+impl FromStr for ThriftProtocol {
+    type Err = ErrorCode;
+
+    fn from_str(s: &str) -> Result<ThriftProtocol> {
+        let s = s.to_lowercase();
+        match s.as_str() {
+            "binary" => Ok(ThriftProtocol::Binary),
+            _ => Err(ErrorCode::StorageOther("invalid thrift protocol spec")),
+        }
+    }
+}
+
+impl Display for ThriftProtocol {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Binary => write!(f, "binary"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct HiveCatalogConfig {
+    pub meta_store_address: String,
```

Comment:
How about `address`?

Reply:
Let's keep this, I found `serde` or `clap` will fail on conflicted namings.